numpy/core/include/numpy/npy_common.h

```diff
@@ -61,6 +61,13 @@
 #define NPY_UNLIKELY(x) (x)
 #endif
 
+#ifdef HAVE___BUILTIN_PREFETCH
+/* unlike _mm_prefetch also works on non-x86 */
+#define NPY_PREFETCH(x, rw, loc) __builtin_prefetch((x), (rw), (loc))
+#else
+#define NPY_PREFETCH(x, rw,loc)
```

Comment:
Any reason not to try `_mm_prefetch` as a fallback? I suppose it would need detection also.